sdk/src/contexts/solana/utils/utils/transaction.ts

```diff
@@ -206,3 +142,71 @@ export async function sendAndConfirmTransactionsWithRetry(
 
   return Promise.resolve(output);
 }
+
+// This function signs and sends the transaction while constantly checking for confirmation
+// and resending the transaction if it hasn't been confirmed after the specified interval
+// NOTE: The caller is responsible for simulating the transaction and setting any compute budget
+// or priority fee before calling this function
+// See https://docs.triton.one/chains/solana/sending-txs for more information
+export async function signSendAndConfirmTransaction(
+  connection: Connection,
+  signTransaction: SignTransaction,
+  payer: PublicKeyInitData,
+  unsignedTransaction: Transaction,
+  commitment: Commitment = 'finalized',
+  txRetryInterval = 5000,
+): Promise<TransactionSignatureAndResponse> {
+  const { blockhash, lastValidBlockHeight } =
+    await connection.getLatestBlockhash({
+      commitment,
+    });
+  unsignedTransaction.recentBlockhash = blockhash;
+  unsignedTransaction.feePayer = new PublicKey(payer);
+  const tx = await signTransaction(unsignedTransaction);
+  let confirmTransactionPromise: Promise<
+    RpcResponseAndContext<SignatureResult>
+  > | null = null;
+  let confirmedTx: RpcResponseAndContext<SignatureResult> | null = null;
+  let txSendAttempts = 1;
+  let signature = '';
+  const serializedTx = tx.serialize();
+  const sendOptions = {
+    skipPreflight: true,
+    maxRetries: 0,
+    preFlightCommitment: commitment, // See PR and linked issue for why setting this matters: https://github.com/anza-xyz/agave/pull/483
+  };
+  signature = await connection.sendRawTransaction(serializedTx, sendOptions);
+  confirmTransactionPromise = connection.confirmTransaction(
+    {
+      signature,
+      blockhash,
+      lastValidBlockHeight,
+    },
+    commitment,
+  );
+  // Keep checking for confirmation until the promise resolves or the timeout is reached
+  while (!confirmedTx) {
```

Comment:
this will retry indefinitely. do you think we should still give up after some amount of time, to avoid an infinite loading spinner? for example what if there's a network outage? 

also, what if the transaction throws an error? wouldn't we be spending the user's gas on infinite retries?

Reply:
This will retry until the call to `confirmTransaction` throws, e.g. if the tx isn't confirmed after the block height expires (`TransactionExpiredBlockheightExceededError`)

Transaction errors should be caught during simulation. The TX isn't resigned, we just broadcast the same signed TX repeatedly until it's confirmed or not

Reply:
I see this follows Triton's example here: https://github.com/rpcpool/optimized-txs-examples/blob/main/jupiterSwap.mjs#L147

The `while`/`break` behavior + Promises created in a loop make me a bit nervous, would be good to double check this can't result in an infinite loop or unsettled promises

Reply:
The `while` loop won't result in an infinite loop because it breaks when the tx is confirmed. If the tx is not confirmed because the block height is exceeded, then `confirmTransactionPromise` will throw an exception and break the loop.